Mathlib/MeasureTheory/VectorMeasure/WithDensity.lean

```diff
@@ -123,15 +123,17 @@ theorem withDensityᵥ_smul_eq_withDensityᵥ_withDensity {f : α → ℝ≥0} {
   rw [withDensityᵥ_apply hfg hs,
     withDensityᵥ_apply ((integrable_withDensity_iff_integrable_smul₀ hf).mpr hfg) hs,
     setIntegral_withDensity_eq_setIntegral_smul₀ hf.restrict _ hs]
-  rfl
+  simp only [Pi.smul_apply']
 
 theorem withDensityᵥ_smul_eq_withDensityᵥ_withDensity' {f : α → ℝ≥0∞} {g : α → E}
     (hf : AEMeasurable f μ) (hflt : ∀ᵐ x ∂μ, f x < ∞)
     (hfg : Integrable (fun x ↦ (f x).toReal • g x) μ) :
     μ.withDensityᵥ (fun x ↦ (f x).toReal • g x) = (μ.withDensity f).withDensityᵥ g := by
   rw [← withDensity_congr_ae (coe_toNNReal_ae_eq hflt),
     ← withDensityᵥ_smul_eq_withDensityᵥ_withDensity hf.ennreal_toNNReal hfg]
-  rfl
+  apply congr_arg
+  ext
+  simp [NNReal.smul_def, ENNReal.coe_toNNReal_eq_toReal]
```

Comment:
Also not sure this is an improvement

Reply:
Probably the slowdown is due to this `simp`. The slowdown in Haar.Unique is probably also due to the `simp` but more reasonable. (Files beyond the first two in the table below are not touched by this PR.)
![image](https://github.com/user-attachments/assets/ce32a033-2f0c-414b-8de4-270ffdeaca8f)